Doc/c-api/structures.rst

```diff
@@ -561,9 +561,9 @@ The following flags can be used with :c:member:`PyMemberDef.flags`:
    :c:member:`PyMemberDef.offset` to the offset from the ``PyObject`` struct.
 
 .. index::
-   single: READ_RESTRICTED
-   single: WRITE_RESTRICTED
-   single: RESTRICTED
+   single: READ_RESTRICTED (C macro)
```

Comment:
Just wondering... would it make sense to indicate in the index that these `*_RESTRICTED` macros are deprecated? For example:

```
   single: READ_RESTRICTED (deprecated C macro)
   single: WRITE_RESTRICTED (deprecated C macro)
   single: RESTRICTED (deprecated C macro)
```

If so, this might well be applied in many other places.

Reply:
If we add
```rst
.. c:macro:: READ_RESTRICTED
```
it will add index entry for "READ_RESTRICTED (C macro)". No way to add "deprecated" here (unless we write our own Sphinx extensions). And some of deprecated names are declared in such way, so we will have inconsistency,

Also, I think that an index entry is just an index entry. It contains the name and maybe some qualifier to distinguish it from the same name in other namespaces. But there is no non-deprecated C macro with the same name.

Reply:
These index entries were added manually, right? Can't they just be edited right where they sit? That way, the word "deprecated" will be part of the index entry, subtly discouraging people from using them in their (new) code. I wasn't thinking of adding a `c:macro` directive.

Reply:
But if we have two deprecated names, and one of them is declared with with the `c:macro` directive, the index entries will be:
```
BAR (C macro)
BAZ (deprecated C macro)
```
Both names are deprecated, but only one of them has an index entry that says this. Would not it confuse users?

I think that the index is not a right place for this. It is just an index, and you have to click the link to get the information.

Reply:
That's fine. Was just tossing the idea out there.